.github/workflows/test_and_release.yml

```diff
@@ -0,0 +1,120 @@
+name: Test and Release
+
+on:
+  workflow_dispatch:
+  release:
+    types: [published]
+  pull_request:
+    branches:
+      - master
+      - v[0-9]+.[0-9]+-dev
+
+jobs:
+  test:
+    name: Run wallet-lib tests
+    runs-on: ubuntu-20.04
+    timeout-minutes: 60
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: '12'
+
+      - name: Enable NPM cache
+        uses: actions/cache@v2
+        with:
+          path: '~/.npm'
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
+
+      - name: Cache Docker layers
+        uses: satackey/action-docker-layer-caching@v0.0.11
+        continue-on-error: true
+        with:
+          concurrency: 8
+
+      - name: Install NPM dependencies
+        run: npm ci
+
+      - name: Create dotenv file
+        run: cp .env.example .env
+
+      - uses: dashevo/gh-action-runtime-vars@master
+        id: runtime-vars
+        if: ${{ contains(github.event.pull_request.body, '/set-var')}}
+        with:
+          pull-request-body: ${{ github.event.pull_request.body }}
+
+      - uses: dashevo/gh-action-platform-branches@master
+        id: platform-branches
+        with:
+          override-major-version: 0
+          override-dashmate-branch: ${{ steps.runtime-vars.outputs.dashmate-branch }}
+          override-testsuite-branch: ${{ steps.runtime-vars.outputs.testsuite-branch }}
+
+      - uses: dashevo/gh-action-start-local-network@master
+        with:
+          actions-cache-url: ${{ steps.platform-branches.outputs.cache-url }}
+          actions-cache-token: ${{ steps.platform-branches.outputs.cache-token }}
+          drive-branch: ${{ steps.runtime-vars.outputs.drive-branch }}
+          dapi-branch: ${{ steps.runtime-vars.outputs.dapi-branch }}
+          dashmate-branch: ${{ steps.platform-branches.outputs.dashmate-branch }}
+          install-sdk-version: ${{ steps.runtime-vars.outputs.install-sdk-version }}
+
+      - name: Parametrize environment
+        run: |
+          echo "DAPI_SEED=127.0.0.1" > .env
+          echo "FAUCET_PRIVATE_KEY=${{ env.faucet-private-key }}" >> .env
+          echo "NETWORK=regtest" >> .env
+          echo "DPNS_CONTRACT_ID=${{ env.dpns-contract-id }}" >> .env
+
+      - name: Check NPM package lock version is updated
```

Comment:
I think the efficient order will be:
1. enable NPM cache
2. install NPM deps
3. Run check package
4. run linter
5. docker cache
5. dashevo/gh-action-runtime-vars@master
6. dashevo/gh-action-platform-branches@master
7.  dashevo/gh-action-start-local-network@master
8. Create .env file
9. Run tests
10. show docker